playbooks/configure-apt-sources.yml

```diff
@@ -46,30 +46,43 @@
       set_fact:
         rpc_openstack: "{{ ansible_local['rpc_openstack']['rpc_artifacts'] }}"
 
-    - name: Set the rpc-artifact variables
+    - name: Set apt_artifact_enabled (based on previous settings)
       set_fact:
         apt_artifact_enabled: "{{ rpc_openstack['apt_artifact_enabled'] }}"
       when:
         - rpc_openstack['apt_artifact_enabled'] is defined
         - apt_artifact_enabled is undefined
 
+    - name: Set apt_artifact_mode (based on previous settings)
+      set_fact:
+        apt_artifact_mode: "{{ rpc_openstack['apt_artifact_mode'] }}"
+      when:
+        - rpc_openstack['apt_artifact_mode'] is defined
+        - apt_artifact_mode is undefined
+
     - name: Check for artifacts
       uri:
         url: "{{ rpco_mirror_base_url }}/apt-mirror/integrated/dists/{{ rpc_release }}-{{ ansible_distribution_release }}/Release"
         method: HEAD
       failed_when: false
       register: check_artifacts
 
-    - name: Set artifacts found
+    - name: Set apt_artifact_found
       set_fact:
         apt_artifact_found: "{{ check_artifacts.status == 200 }}"
 
-    - name: Set artifacts enabled
+    - name: Set apt_artifact_enabled (based on whether they exist)
       set_fact:
         apt_artifact_enabled: "{{ apt_artifact_found | bool }}"
       when:
         - apt_artifact_enabled is undefined
 
+    - name: Set apt_artifact_mode default value
+      set_fact:
+        apt_artifact_mode: "strict"
+      when:
+        - apt_artifact_mode is undefined
+
```

Comment:
We should add a failure condition when the "apt_artifact_mode" is not set to "strict" or "loose". From an operator POV, the system should provide feedback to combat against typos especially when with an option that may impact the entire deployment.

Reply:
It's not really necessary. All the conditionals test against the default of 'strict'. If the value is anything else, it considers it loose.

Reply:
I'd rather this error on the side of caution. I can imagine a scenario where someone wants strict mode but typos it in some way, like using a capital letter somehow, resulting in a deployment in loose mode unexpectedly.

Reply:
Alright, once I'm back from holiday I can do that. Or we can merge this and a patch can go on top of it to evolve it.

Reply:
Done.